extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeStreamExtractor.java

```diff
@@ -497,67 +501,112 @@ public String getHlsUrl() throws ParsingException {
         try {
             return playerResponse.getObject("streamingData").getString("hlsManifestUrl");
         } catch (final Exception e) {
-            throw new ParsingException("Could not get hls manifest url", e);
+            throw new ParsingException("Could not get HLS manifest url", e);
         }
     }
 
     @Override
     public List<AudioStream> getAudioStreams() throws ExtractionException {
         assertPageFetched();
         final List<AudioStream> audioStreams = new ArrayList<>();
+        final StreamType streamType = this.getStreamType();
 
         try {
             for (final Map.Entry<String, ItagItem> entry : getItags(ADAPTIVE_FORMATS, ItagItem.ItagType.AUDIO).entrySet()) {
                 final ItagItem itag = entry.getValue();
-                final AudioStream audioStream = new AudioStream(entry.getKey(), itag);
-                if (!Stream.containSimilarStream(audioStream, audioStreams)) {
-                    audioStreams.add(audioStream);
+                final AudioStream audioStream;
+                if (streamType == StreamType.VIDEO_STREAM) {
```

Comment:
Could you add a comment explaining this `if`? Are you just checking if this video is a live or not?
Also in the other places